bridge_ui/src/components/Transactions.tsx

```diff
@@ -0,0 +1,519 @@
+import {
+  ChainId,
+  CHAIN_ID_ALEPHIUM,
+  CHAIN_ID_ETH,
+  getIsTransferCompletedEth
+} from "@certusone/wormhole-sdk";
+import {
+  Box,
+  Card,
+  CircularProgress,
+  Collapse,
+  Container,
+  List,
+  ListItem,
+  ListItemText,
+  makeStyles,
+  MenuItem,
+  TextField,
+  Typography
+} from "@material-ui/core";
+import { ExpandLess, ExpandMore } from "@material-ui/icons";
+import { useCallback, useEffect, useState } from "react";
+import { COLORS } from "../muiTheme";
+import {
+  ALEPHIUM_BRIDGE_ADDRESS,
+  ALEPHIUM_EXPLORER_HOST,
+  ALEPHIUM_MINIMAL_CONSISTENCY_LEVEL,
+  ALEPHIUM_TOKEN_BRIDGE_CONTRACT_ID,
+  CHAINS,
+  CHAINS_WITH_NFT_SUPPORT,
+  getTokenBridgeAddressForChain
+} from "../utils/consts";
+import { Transaction, TransactionDB, TxStatus } from "../utils/db";
+import ChainSelect from "./ChainSelect";
+import KeyAndBalance from "./KeyAndBalance";
+import { useLiveQuery } from "dexie-react-hooks"
+import { NodeProvider, ExplorerProvider, explorer } from "@alephium/web3"
+import { isAlphTxConfirmed, isAlphTxNotFound } from "../utils/alephium";
+import { useAlephiumWallet } from "../contexts/AlephiumWalletContext";
+import useIsWalletReady from "../hooks/useIsWalletReady";
+import { getSignedVAAWithRetry } from "../utils/getSignedVAAWithRetry";
+import { useEthereumProvider } from "../contexts/EthereumProviderContext";
+import { Alert } from "@material-ui/lab";
+import ButtonWithLoader from "./ButtonWithLoader";
+
+const useStyles = makeStyles((theme) => ({
+  mainCard: {
+    padding: "32px 32px 16px",
+    backgroundColor: COLORS.whiteWithTransparency,
+  },
+  list: {
+    width: '100%',
+    backgroundColor: theme.palette.background.paper,
+  },
+  nested: {
+    paddingLeft: theme.spacing(4),
+  },
+  loader: {
+    marginLeft: theme.spacing(50)
+  },
+  error: {
+    marginTop: theme.spacing(1),
+    textAlign: "center",
+  },
+  info: {
+    marginTop: theme.spacing(1),
+  },
+}));
+
+async function getTxsByStatus(
+  status: TxStatus,
+  sourceChainId: ChainId,
+  targetChainId: ChainId
+): Promise<Transaction[]> {
+  return TransactionDB.getInstance().txs
+    .where({
+      "status": status,
+      "sourceChainId": sourceChainId,
+      "targetChainId": targetChainId
+    })
+    .toArray()
+}
+
+interface TxIdAndSequence {
+  txId: string
+  sequence: string
+}
+
+async function getBridgeTransferTxInfo(
+  nodeProvider: NodeProvider,
+  tx: explorer.Transaction
+): Promise<TxIdAndSequence | undefined> {
+  if (typeof tx.inputs === 'undefined' || tx.inputs.length === 0) {
+    return undefined
+  }
+  const haveContractInput = tx.inputs.some(input => typeof input.unlockScript === 'undefined')
+  if (!haveContractInput) {
+    return undefined
+  }
+  const events = await nodeProvider.events.getEventsTxIdTxid(tx.hash)
```

Comment:
I created a Github issued to support events on explorer-backend.

Reply:
Thanks!